generators/pkg/writer/write.go

```diff
@@ -22,17 +22,39 @@ func WriteToDisk(filename string, writer io.Writer) error {
 	}
 	defer file.Close()
 
-	spec, err := parser.Parse(file)
+	github, err := parser.Parse(file)
 	if err != nil {
 		return fmt.Errorf("unable to parse the file from %s: %w", filename, err)
 	}
 
-	task := generator.GenerateTask(spec)
-	data, err := yaml.Marshal(&task)
-	if err != nil {
-		return fmt.Errorf("unable to marshal the file from %s: %w", filename, err)
+	pipeline, tasks, err := generator.GeneratePipeline(github)
+
+	// write tasks
+	for _, task := range tasks {
+		_, err = writer.Write([]byte("---\n"))
+		if err != nil {
+			return err
+		}
```

Comment:
```suggestion
		if _, err = writer.Write([]byte("---\n")); err != nil {
			return err
		}
```

Here and elsewhere in this file